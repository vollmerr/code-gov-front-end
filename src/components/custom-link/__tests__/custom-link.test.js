import React from 'react';
import { Link } from 'react-router-dom'

import { BrowserRouter as Router } from 'react-router-dom';
import CustomLink from 'components/custom-link';
import { mount } from 'enzyme';

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares);

describe('Testing <CustomLink /> component', () => {
  let wrapper;
  const props = {
    children: 'test child',
    to: '/browse',
    className: 'test',
    updateStore: jest.fn(),
    onClick: jest.fn()
  }
  const store = mockStore(props);
  beforeEach(() => {
    wrapper = mount(<Router><CustomLink {...props} store={store} /></Router>);
  });
  it('tests if the component renders', () => {
    expect(wrapper.exists()).toBe(true);
  });
  it('renders 1 CustomLink without children', () => {
    const noChildState = {
      className: 'test-className',
      title: 'test title',
      to: 'test to',
      updateStore: jest.fn(),
    }
    wrapper = mount(<Router><CustomLink {...noChildState} store={store}/></Router>);
    expect(wrapper.exists()).toBe(true);
  });
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('testing onClick with props onClick', () => {
    wrapper.find(Link).simulate('click');
    expect(props.onClick).toHaveBeenCalled();
  });
  it('testing updateStore with /search', () => {
    const props = {
      children: 'test child',
      to: '/search',
      className: 'test',
      updateStore: jest.fn(),
      onClick: jest.fn()
    }
    const store = mockStore(props);
    wrapper = mount(<Router><CustomLink {...props} store={store} /></Router>);
    wrapper.find(Link).simulate('click');
    expect(props.onClick).toHaveBeenCalled();
  });
  it('testing updateStore with "/open-tasks"', () => {
    const props = {
      children: 'test child',
      to: '/open-tasks',
      className: 'test',
      updateStore: jest.fn(),
      onClick: jest.fn()
    }
    const store = mockStore(props);
    wrapper = mount(<Router><CustomLink {...props} store={store} /></Router>);
    wrapper.find(Link).simulate('click');
    expect(props.onClick).toHaveBeenCalled();
  });
})
