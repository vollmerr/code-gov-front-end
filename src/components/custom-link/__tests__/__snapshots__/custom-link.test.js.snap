// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <CustomLink /> component should render correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/home",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Connect(CustomLinkComponent)
      className="test"
      onClick={[MockFunction]}
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
      to="/browse"
      updateStore={[MockFunction]}
    >
      <CustomLinkComponent
        className="test"
        onClick={[MockFunction]}
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
        to="/browse"
        updateStore={[Function]}
      >
        <Link
          className="test"
          onClick={[Function]}
          replace={false}
          to="/browse"
        >
          <a
            className="test"
            href="/browse"
            onClick={[Function]}
          >
            test child
          </a>
        </Link>
      </CustomLinkComponent>
    </Connect(CustomLinkComponent)>
  </Router>
</BrowserRouter>
`;
